### DISCUSSION TITLE
Pre-rendering multiple files

### DISCUSSION THREAD

**Discussion by BradySullivan1** [quarto-dev/quarto-web#1258] [Q&A]
Hello, I am using the pre-render function in the project yml. I would like to pre render multiple files but am running into errors. What would the syntax for prerendering multiple files?

**Comment by cderv** [comment]
> I would like to pre render multiple files but am running into errors. What would the syntax for prerendering multiple files?

What did you try exactly ? Can you share more ?

**Reply by BradySullivan1** [reply]
I have found a better way without having to pre-render multiple files. I am using the conditional content syntax (::: {.content-visible etc.}). I am now running into the issue with getting a syntax error saying there is an unclosed bracket, but I do not see one. This is the chunk that is giving me the error.

 ::: {.content-visible when-meta= ((param.a in on_line) & is_x) }
    {{< include file.qmd >}}
:::

SyntaxError: '[' was never closed (<string>, line 1)

**Reply by cderv** [reply]
I am surprised about the usage of `when-meta`. Did you find this example like this somewhere ? 

`when-meta` works with boolean values in Lua : https://quarto.org/docs/authoring/conditional.html#matching-against-metadata

What are you trying to do with `((param.a in on_line) & is_x` ? Where are `on_line` and `is_x` defined ? 

I don't think this will work - definitely not documented and expected.

**Reply by BradySullivan1** [reply]
I am trying to conditional show the .qmd file depending on if param.a is in the function on_line and is_x is true. So show the .qmd on the condition that both are true. Both on_line and is_x are defined in a previous code chunk. Is there be a better way to do this?

**Reply by cderv** [reply]
I think there is confusion there. `when-meta` is used as a [Div](https://quarto.org/docs/authoring/markdown-basics.html#sec-divs-and-spans) attribute - this is special Markdown syntax, and not the same as computation cell. 

`when-meta` value won't be evaluated in a computation language by the engine - it will be retrieved from Pandoc's Metadata in Lua filter to check if value is true or false. In our doc, the only example we give is to retrieve a value passed as Metadata in YAML. We don't show example with any language because it does work that way; 

> Both on_line and is_x are defined in a previous code chunk.

Those name will mean nothing when `when-meta` is handled in LUA. `&` does not mean either the and operator in Lua. So you need to understand how all this works in order to find a solution. 

> I am trying to conditional show the .qmd file depending on if param.a is in the function on_line and is_x is true.

It seem you are trying to conditionally show content based on YAML values **AND** other values from cells computations. This is quite advanced usage; 

I believe we have already some discussion about all this. So you can search history and find valuable example. 

Basically, two solution currently are 

- Use computation to create or not the Markdown content you expect - with engine you can produce raw output from a cell https://quarto.org/docs/computations/execution-options.html#raw-output
- Use computation to calculate some values that you need to pass to Quarto rendering processing as a single resulting metadata value for `when-meta` to access. With code cell, you can produce raw output as mentioned which could be a YAML metadata block 
	````yaml
	---
	custom-value: true
	---
	````
	We plan to add helper for that in our R package for example (https://github.com/quarto-dev/quarto-r/issues/137)
	
Hopefully this give you enough information to try out different solution, and learn more how Quarto works regarding computation and conversion through Pandoc.

**Reply by BradySullivan1** [reply]
Thank you! I have tested with an earlier conditional cell: 
::: {.content-visible when-meta= (params.y=="2AD")}

    {{< include _file1.qmd >}}

:::

This works as intended. The idea I have now is to take the other functions and put them in a separate python script which would then turn them into a boolean value and put that value in a variable in the _variables.yml file which I should be able to do what I have above.

**Reply by cscheid** [reply]
```
::: {.content-visible when-meta= (params.y=="2AD")}
```

> This works as intended.

Just to clarify, I don't see how this could possibly be working as intended given the implementation of `when-meta`:

https://github.com/quarto-dev/quarto-cli/blob/8ba05ff2ae45c88a1d619190396441a5025e0e2f/src/resources/filters/customnodes/content-hidden.lua#L156-L167
